pumps/prometheus.go

```diff
@@ -160,30 +162,55 @@ func (p *PrometheusPump) Init(conf interface{}) error {
 	}
 
 	//first we init the base metrics
-	for _, metric := range p.allMetrics {
-		metric.aggregatedObservations = p.conf.AggregateObservations
-		errInit := metric.InitVec()
-		if errInit != nil {
-			p.log.Error(errInit)
-		}
-	}
+	p.initBaseMetrics()
 
 	//then we check the custom ones
 	p.InitCustomMetrics()
 
+	disabledMetrics := p.disableExcludedMetrics()
+
 	p.log.Info("Starting prometheus listener on:", p.conf.Addr)
 
 	http.Handle(p.conf.Path, promhttp.Handler())
 
 	go func() {
 		log.Fatal(http.ListenAndServe(p.conf.Addr, nil))
 	}()
-	p.log.Info(p.GetName() + " Initialized")
+	p.log.WithField("disabled_metrics", disabledMetrics).Info(p.GetName() + " Initialized")
 
 	return nil
 }
 
-//InitCustomMetrics initialise custom prometheus metrics based on p.conf.CustomMetrics and add them into p.allMetrics
+func (p *PrometheusPump) disableExcludedMetrics() []string {
+	//configure base or custom metrics as disabled if needed. This disables exposition entirely during scrapes.
+	var disabledMetrics []string
+	{
+		toDisableSet := map[string]struct{}{}
+		for _, metric := range p.conf.DisabledMetrics {
+			toDisableSet[metric] = struct{}{}
+		}
+
+		for _, metric := range p.allMetrics {
+			if _, ok := toDisableSet[metric.Name]; ok {
+				metric.enabled = false
+				disabledMetrics = append(disabledMetrics, metric.Name)
+			}
+		}
+	}
+	return disabledMetrics
+}
+
+func (p *PrometheusPump) initBaseMetrics() {
+	for _, metric := range p.allMetrics {
```

Comment:
@hsyed-dojo why don't we check and disable the metrics directly inside this method and before initialization? It doesn't make a lot of sense init the metric if we are disabling it right after IMO.

Reply:
The approach I took also applies to any custom metrics that might be configured.

@tbuchaillot The way it is currently implemented we have support to disable any custom metrics that might have been configured. This could be useful for temporarily disabling a metric. 

WDYT ? Happy to apply your suggestion if that is preferable.

Reply:
yes, for sure we have to be clear and avoid any point of confusion. Why not only remove the metric from the config file? am I missing something?